### PR TITLE
chore: change fluid to fullWidth

### PR DESCRIPTION
**Pull request by laurieontech**
This API change better reflects functionality. And it points people to constrained which is what they want in most cases.

TODO:
- remove compat or update it
- rerun snapshots
- needs to go in after #28985 so it can be merged cleanly

[ch22725]

### CODE REVIEW

File: packages/gatsby-plugin-image/README.md [hunk 1]
```diff
@@ -288,13 +288,13 @@ In Gatsby's GraphQL implementation, you specify the type of image with the `layo
 
 # GraphQL resolver
 
-We have added a new `gatsbyImageData` resolver to the `ImageSharp` node. Unlike the existing `fixed` and `fluid` resolvers, this returns a
+We have added a new `gatsbyImageData` resolver to the `ImageSharp` node. Unlike the existing `fixed` and `fullWidth` resolvers, this returns a
```
Comment (ascorbic) [inline packages/gatsby-plugin-image/README.md:291]:
```suggestion
We have added a new `gatsbyImageData` resolver to the `ImageSharp` node. Unlike the existing `fixed` and `fluid` resolvers, this returns a
```
This is referring to the old resolver

File: packages/gatsby-plugin-image/README.md [hunk 2]
```diff
@@ -289,12 +289,12 @@ In Gatsby's GraphQL implementation, you specify the type of image with the `layo
...
 
 We have added a new `gatsbyImageData` resolver to the `ImageSharp` node. Unlike the existing `fixed` and `fluid` resolvers, this returns a
-JSON type, meaning you don't specify the individual fields, but are instead given the whole object. This is because the object is then passed in to the `<GatsbyImage>` component. The API is like this:
+JSON type, meaning you don't specify the individual fields, but are instead given the whole object. This is because the object is then passed into the `<GatsbyImage>` component. The API is like this:
 
 ```graphql
 coverImage: file(relativePath: { eq: "plant.jpg" }) {
   childImageSharp {
-    gatsbyImageData(width: 720, layout: CONSTRAINED, placeholder: TRACED_SVG)
+    gatsbyImageData(maxWidth: 720, layout: CONSTRAINED, placeholder: TRACED_SVG)
```
Comment (ascorbic) [inline packages/gatsby-plugin-image/README.md:297]:
```suggestion
    gatsbyImageData(width: 720, layout: CONSTRAINED, placeholder: TRACED_SVG)
```

File: packages/gatsby-plugin-image/README.md [hunk 3]
```diff
@@ -326,9 +326,10 @@ These arguments can be passed to the `gatsbyImageData()` resolver:
 - **layout**: The layout for the image.
   - `CONSTRAINED`: (default) Resizes to fit its container, up to a maximum width, at which point it will remain fixed in size.
   - `FIXED`: A static image size, that does not resize according to the screen width
-  - `FLUID`: The image resizes to fit its container. Pass a "sizes" option if it isn't going to be the full width of the screen.
+  - `FULL_WIDTH`: The image resizes to fit its container. Pass a "sizes" option if it isn't going to be the full width of the screen.
 - **outputPixelDensities**: A list of image pixel densities to generate, for high-resolution (retina) screens. It will never generate images larger than the source, and will always include a 1x image.
-  Default is `[ 0.25, 0.5, 1, 2 ]`, for fluid/constrained images, and `[ 1, 2 ]` for fixed.
+  Default is `[ 0.25, 0.5, 1, 2 ]`, for fullWidth/constrained images, and `[ 1, 2 ]` for fixed.
+  > > > > > > > master
```
Comment (ascorbic) [inline packages/gatsby-plugin-image/README.md:332]: Conflict here

Comment (laurieontech) [reply packages/gatsby-plugin-image/README.md:332]: AH miss the bottom. Thanks

**Review by ascorbic — APPROVED** [review]
🚀

**Pull request merged**